### PR TITLE
Test App: StoreTest added to test Performance of PersistentStore and Dictionary plugin via JSONRPC, ReST and COMRPC channels

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: examples/StoreTest/StoreTest.cpp [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+
+private:
+    void Operational(const bool upAndRunning)
+    {
+        printf("Operational state of PersistentStore: %s\n", upAndRunning ? _T("true") : _T("false"));
+    }
+};
+
+namespace JSONRPC {
+class PersistentStore : public Thunder::JSONRPC::SmartLinkType<Thunder::Core::JSON::IElement> {
```
Comment (MFransen69) [inline examples/StoreTest/StoreTest.cpp:174]: If we are going to put this in examples people will consider this code that reflects correctly how things should be done and will copy this code as a starting point. So either we will make it so and use JSONRPC::LinkType (see the jsonrpocclient example) and certainly not JsonObject (which is not allowed in ML code) or we just put this code next to our unittest and other test code. (perhaps the latter is a better approach all together). I think it is more of a test then an example, it is also called StoreTest :) so I guess better to out it with the tests

File: examples/StoreTest/StoreTest.cpp [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+template <typename STORE>
+class StorePerformance {
+
+public:
+    StorePerformance(uint32_t waitTime, const Thunder::Core::NodeId& nodeId, const string& callSign)
+        : _store(waitTime, nodeId, callSign)
+        , _callSign(callSign)
+    {
+    }
+    virtual ~StorePerformance() = default;
```
Comment (MFransen69) [inline examples/StoreTest/StoreTest.cpp:395]: why is this virtual? Perhaps I missed it but there does not seem to be any virtual method

File: examples/StoreTest/StoreTest.cpp [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+private:
+    using BaseClass = Thunder::RPC::SmartInterfaceType<Thunder::Exchange::IDictionary >;
+public:
+    Dictionary(const uint32_t waitTime, const Thunder::Core::NodeId& node, const string& callsign)
+        : BaseClass()
+    {
+        BaseClass::Open(waitTime, node, callsign);
+        BaseClass::Aquire<Thunder::Exchange::IDictionary>(waitTime, node, callsign);
+    }
+    ~Dictionary()
```
Comment (MFransen69) [inline examples/StoreTest/StoreTest.cpp:86]: nitpicking I guess, but it should be overriden, please also check below (if we put it in examples it should be rather correct)

File: examples/StoreTest/StoreTest.cpp [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+            string ns = "/" + nameSpace;
+            result = impl->Set(ns, key, value);
+            impl->Release();
+        }
+
+        return (result);
+    }
+
+private:
+    void Operational(const bool upAndRunning)
```
Comment (MFransen69) [inline examples/StoreTest/StoreTest.cpp:120]: think this is an override as well (also check below)

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: examples/StoreTest/StoreTest.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp:174]: Ok, Will move this to unit tests folder

File: examples/StoreTest/StoreTest.cpp (hunk 2 above)
Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp:395]: will correct

File: examples/StoreTest/StoreTest.cpp (hunk 3 above)
Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp:86]: will check

File: examples/StoreTest/StoreTest.cpp (hunk 4 above)
Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp:120]: will check

File: examples/StoreTest/StoreTest.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp:174]: I changed JsonObject to JSON::Container based class based on Parameter and Response requirement.
I used JSONRPC::SmartLinkType instead JSONRPC::LinkType to get the plugin activated/deactivated status to return proper states from IsOperational() method.

Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp:174]: Will move this to Thunder as next step :)

Comment (HaseenaSainul) [reply examples/StoreTest/StoreTest.cpp]: Moved StoreTest to tests

**Review by MFransen69 — APPROVED** [review]

**Pull request merged**